### ISSUE TITLE
Can we make private use subtags constructor always safe?

### ISSUE THREAD

**Issue by sffc** [unicode-org/icu4x#4149]
We have the function `icu::locid::extensions::private::Private::from_vec_unchecked`, but it's not clear to me which invariants need to be upheld for that function to be called "unchecked". Can we just name it `from_vec`?

https://unicode-org.github.io/icu4x/docs/icu_locid/extensions/private/struct.Private.html#method.from_vec_unchecked

@zbraniecki

**Comment by zbraniecki** [comment]
I'm scrambling a bit but my guess would be that we don't check for duplications or order.

**Comment by sffc** [comment]
Do private use subtags need to be ordered and unique?

**Comment by sffc** [comment]
Seems like this is available to work on; just check the spec to make sure that the change is conformant with it.

**Comment by zbraniecki** [comment]
My understanding of https://unicode.org/reports/tr35/tr35.html#Canonical_Unicode_Locale_Identifiers is that Private is not excluded from canonicalization - duplication, order, casing,

**Comment by sffc** [comment]
OK, let's just improve the docs then.

**Comment by sffc** [comment]
> My understanding of https://unicode.org/reports/tr35/tr35.html#Canonical_Unicode_Locale_Identifiers is that Private is not excluded from canonicalization - duplication, order, casing,

The spec lists the specific conditions that must be met in order for a locale to be in canonical syntax, and private use subtag order is not listed?

**Comment by zbraniecki** [comment]
I agree. duplicate subtags are allowed, sorting is not required, canonical casing is lower case.
